### ISSUE TITLE
Options don't show up for metro style apps.

### ISSUE THREAD

**Issue by jomply** [navossoc/MenuTools#2]
Hi

The context menus for Win 10 metro apps (e.g. Edge, Calculator) don't show the added options, is there any chance of adding this in?

Cheers!

**Comment by navossoc** [comment]
To be honest, I don't know if that is possible.
As far I know, the Win32 applications doesn't "talks" with the Universal (store) applications.

I'll look into it too.

**Comment by vitaminj** [comment]
Workaround: they show up if you shift-right-click on the taskbar button

**Comment by SlyAceZeta** [comment]
> 
> 
> Workaround: they show up if you shift-right-click on the taskbar button

YES, thank you! This allows MenuTools to even work for programs that don't have menu bars like Firefox or Steam. This still works as of Windows 10 v1909 and MenuTools 1.0.2. Thank you so much!!

**Comment by Darthagnon** [comment]
Workaround and regular doesn't work for [Process Hacker](https://processhacker.sourceforge.io/downloads.php)

**Comment by fade2gray** [comment]
Was searching for a method to minimize VirtualBox VMs to the tray and found [this post](https://forums.virtualbox.org/viewtopic.php?f=2&t=94355) which led me here.

> ... I then tried MenuTools ..., and it DOES work ...
> ... But it doesn't work for Firefox ...

Well, it **does** work for Firefox as outlined by @vitaminj , but I **can't** get it to work for VMs.

Does anyone know of a method that works with VMs?

Thanks

**Comment by ayunami2000** [comment]
You can hold shift and right click the icon in the taskbar:
![image](https://user-images.githubusercontent.com/24785136/94028023-7d22c900-fd89-11ea-81a7-a388e4209f6a.png)

**Comment by fade2gray** [comment]
@ayunami2000 

The Menu Tools options **are _not_ shown** for VirtualBox virtual machine taskbar icons when using Shift+RightClick.

![Capture](https://user-images.githubusercontent.com/1332009/94031206-87a78780-fdb6-11ea-9513-a7b02a0c5afb.PNG)

**Comment by ayunami2000** [comment]
Perhaps try shift+right clicking the main virtualbox window? Maybe it'll
affect all virtualbox windows. (just a guess though) My guess is that
virtualbox has a custom handler for that and replaces it with only the
default menu.

When right clicking the title bar of metro apps, it shows a similar menu.
Perhaps there's a second version of the menu, and MenuTools only adds to
one of them?

**Comment by fade2gray** [comment]
@ayunami2000 I've tried every method I can without luck.

**Comment by navossoc** [comment]
Do people still use this? 😕

Let me know, do you guys think we should create a new issue here on github to "brainstorm" a new version of this program with a few enhancements and bug fixes?

I would like to see some ideas first to start planning ahead.

**Comment by joe-muller** [comment]
I use the Alway on Top feature daily, super nifty

**Comment by fade2gray** [comment]
@navossoc 

I found MT when looking for an app to implement a long time feature request of VirtualBox to be able to minimize VMs to the tray.

MT v1.0.2  will not minimize VMs for me in Windows 10, but v1.0.1 will with the caveat that it won't autorun at logon and prompts for UAC permission when run manually - my workaround for this is to...

1. Install MT v1.0.2.
2. Copy and save `C:\Program Files\MenuTools\helper.exe`.
3. Export the `MenuToools Task Schedule`.
4. Uninstall MT v1.0.2 and install v1.0.1.
5. Copy the missing `helper.exe` to `C:\Program Files\MenuTools`.
6. Import the `MenuToools Task Schedule`.

So, if you could fix that in v1.0.3 I'd be very grateful 👍

**Comment by facboy** [comment]
@fade2gray 

it's very odd, at first glance it doesn't look like there are any changes that would cause it not to work with VirtualBox.  i tried building 1.0.1 locally and that didn't seem to work with VirtualBox either, though perhaps I didn't rebuild it properly after building 1.0.2.

**Comment by navossoc** [comment]
My best guess is some isolation issue or maybe UAC itself.

Both processes are running on the same user? with or without administrator privileges?
You can easily check that on process hacker or process explorer.

[]'s

**Comment by facboy** [comment]
with 1.0.1, MenuTools is elevated, VirtualBox is not.  same user.

i didn't check 1.0.2 yet.

**Comment by fade2gray** [comment]
Bit of a selfish request, but would it be possible to implement a right-click menu option for restoring windows from the tray?

Unfortunately, my gaming mouse has developed a twitchy left mouse button when trying to double-click and it will sometimes end up also double-clicking the next tray icon that replaces the one just restored, and I can't afford to replace my mouse at the moment.

**Comment by facboy** [comment]
interesting...are you 100% certain it's your mouse?  i have this problem where the 'next' icon gets triggered as well, but i'm not having 'double-click' problems in anything else.

**Comment by fade2gray** [comment]
Yes, it definitely developed a problem performing multiple clicks when only single-clicking. I managed to alleviate this by increasing the d/c speed in mouse settings, but a single click still does occasionally produce multiple clicks.
